testsuite/tests/typing-layouts/allow_any.ml

```diff
@@ -1,190 +1,166 @@
+let use_uncontended : 'a @ uncontended -> 'a = fun x -> x
+
 (* Baseline: if the jkind doesn't match, we should get an error. *)
-module Mismatched_no_attrs : sig
-  type t : float64
-end = struct
-  type t = string
-end
+type t : value mod uncontended = { mutable contents : string }
 [%%expect{|
-Lines 3-5, characters 6-3:
-3 | ......struct
-4 |   type t = string
-5 | end
-Error: Signature mismatch:
-       Modules do not match:
-         sig type t = string end
-       is not included in
-         sig type t : float64 end
-       Type declarations do not match:
-         type t = string
-       is not included in
-         type t : float64
-       The layout of the first is value
-         because it is the primitive type string.
-       But the layout of the first must be a sublayout of float64
-         because of the definition of t at line 2, characters 2-18.
+val use_as_value : 'a -> 'a = <fun>
+val use_uncontended : ('a : value_or_null). 'a -> 'a = <fun>
+Line 5, characters 0-62:
+5 | type t : value mod uncontended = { mutable contents : string }
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The kind of type "t" is value
+         because it's a boxed record type.
+       But the kind of type "t" must be a subkind of value mod uncontended
+         because of the annotation on the declaration of the type t.
 |}]
 
-(* On the other hand, if we set the correct attributes on both the impl and the intf, we
-   shouldn't get an error (though, obviously, this is completely unsound!) *)
-module Mismatched_with_both_attrs : sig
-  type t : float64
-  [@@unsafe_allow_any_kind_in_impl "I love segfaults"]
-end = struct
-  type t = string
-  [@@unsafe_allow_any_kind_in_intf "I love segfaults"]
-end
+(* On the other hand, if we set the attribute, we shouldn't get an error. *)
+type t : value mod uncontended = { mutable contents : string }
+[@@unsafe_allow_any_mode_crossing]
+let f (x : t @@ contended) = use_uncontended x
 [%%expect{|
-module Mismatched_with_both_attrs : sig type t : float64 end
+type t : value mod uncontended = { mutable contents : string; }
+val f : t @ contended -> t = <fun>
 |}]
 
-(* If we set the attributes but *don't* get a kind mismatch, we ought to be fine *)
-module Matching : sig
-  type t : value
-  [@@unsafe_allow_any_kind_in_impl "I love segfaults"]
-end = struct
-  type t = string
-  [@@unsafe_allow_any_kind_in_intf "I love segfaults"]
-end
+(* If we set the attribute but *don't* get a kind mismatch, we ought to be fine *)
+type t : value mod many portable uncontended = string
+[@@unsafe_allow_any_mode_crossing]
 [%%expect{|
-Lines 2-3, characters 2-54:
-2 | ..type t : value
-3 |   [@@unsafe_allow_any_kind_in_impl "I love segfaults"]
-Warning 212 [unnecessary-allow-any-kind]: [@@allow_any_kind_in_intf] and [@@allow_any_kind_in_impl] set on a
-type, but the kind matches. The attributes can be removed.
-
-module Matching : sig type t end
+type t = string
```

Comment:
This doesn't need to print the attribute, as it was useless here (but it could print it if that's easier).